Automattic/sensei
# Add focus mode

Fixes #4381 

### Changes proposed in this Pull Request

* Add a toggle to collapse sidebar & minimize header
* Add styles & a helper script
* Save & restore collapsed state via sessionStorage
* Update layout — set up fixed sidebar

### Testing instructions

* Enable the feature flag: `add_filter( 'sensei_feature_flag_course_theme', '__return_true' );`
* Open a lesson that has the course theme enabled
* Click `Collapse` on the sidebar. Observe the collapse.
* Reload the page. Observe it staying collapsed.
* Click `Expand`. Observe expanding.

### Notes
* Button placement and style is still subject to change so I haven't added icons yet


### Screenshot / Video

<img width="1263" alt="image" src="https://user-images.githubusercontent.com/176949/143502640-1dae2e2e-425f-4fa6-8a83-703d42c1095f.png">

![QuickTime movie](https://user-images.githubusercontent.com/176949/143502729-d9f46ebd-ab88-46ec-8652-bcb53d2c7f46.gif)
